experimental/GModule/src/Cohomology.jl

```diff
@@ -338,6 +353,23 @@ function Oscar.zero!(a::AbstractAlgebra.FPModuleElem)
   zero!(a.v)
 end
 
+direct_sum_elem_type(::Type{<:FPModule{T}}) where T = Generic.DirectSumModuleElem{T}
+direct_sum_elem_type(::Type{FinGenAbGroup}) = FinGenAbGroupElem
+function direct_sum_elem_type(T::TypeVar) 
+   @show T, typeof(T)
+   Any
+end
+direct_sum_elem_type(::FinGenAbGroup) = FinGenAbGroupElem
+direct_sum_elem_type(::FPModule{T}) where T = Generic.DirectSumModuleElem{T}
```

Comment:
This would be more in line with how we define `parent_type`, base_ring_type` etc. in AA:
```suggestion
direct_sum_elem_type(x) = direct_sum_elem_type(typeof(x))
direct_sum_elem_type(T::DataType) = throw(MethodError(direct_sum_elem_type, (T,)))

direct_sum_elem_type(::Type{<:FPModule{T}}) where T = Generic.DirectSumModuleElem{T}
direct_sum_elem_type(::Type{FinGenAbGroup}) = FinGenAbGroupElem
```